### ISSUE TITLE
[charts] Document how to use animated images in charts' overlay

### ISSUE THREAD

**Issue by TheOneTheOnlyJJ** [mui/mui-x#12990]
### Summary

https://github.com/mui/mui-x/assets/63308694/ab713a88-8ccc-41bd-8e9a-f79620fd9175

Now that we have support for custom overlays (thanks to #12817 being completed), we could go further and add support for animated image overlays.

The formats supported could be GIFs, animated SVGs, and Lottie files. Feel free to advocate for or against these or any other formats.
Lottie files specifically would be very helpful as they allow for programmatic color changes to the image content, and we could leverage the theme and the new color system from MD3 further down the line, especially if the illustrated default animations would be supported out of the box.
Unfortunately, I could not find these specific animated images online, though it would be very convenient to support them out of the box, as they conform to the MD standards and make for a great user experience when used.

### Examples

[Official MD2 docs example](https://kstatic.googleusercontent.com/files/3af67b9d1b746bf98e1356335022674d933272872ce2f7c8a52e412f2d86f98adc1c16d3eca3d152ac0b2bbf845c7a0c86e3beb4924dcdeca9a544f312a5d45e) (the one from above)

### Motivation

I am suggesting this because [such animations are briefly illustrated and advocated for in the official MD2 docs](https://m2.material.io/design/communication/data-visualization.html#behavior), in the last paragraph of the Behaviour section of the Data Visualisation page (just above the [Dashboards](https://m2.material.io/design/communication/data-visualization.html#dashboards) section).

**Comment by michelengelen** [comment]
Hey @TheOneTheOnlyJJ and thanks for raising this feature request.
Thats an interesting feature... let me add this to the board for the team to estimate.

Not sure if we will provide built in animated images for the empty/loading state, but we could at least support them.

WDYT @alexfauquette ?

**Comment by TheOneTheOnlyJJ** [comment]
> Not sure if we will provide built in animated images for the empty/loading state, but we could at least support them.

Yeah, I do not know any place where such ready-made assets with favorable licensing agreements could be found freely.

Support would be a great start, and developers looking to use such animations could get them custom made by art creators, thus being able to also have very specific animations that integrate well with the product. This would make the feature an optional step towards very extensive product-specific custom UX.

However, I believe an effort could be made to contact the Material Design team here regarding these specific animation assets seen in the example linked above.
Maybe they still have the assets saved separately and could provide us with them. Most likely though, even if the assets still exist, licensing problems will arise.
I believe it is worth a try, as the benefits of potentially having these animations as defaults in the library are worth the effort of sending a few e-mails to the MD team.

**Comment by alexfauquette** [comment]
If you want to get this animation of small stick person doing sport in your chart, It's not planed to add it (even though I admit I's a very funny one)

You can already put a [custom overlay](https://mui.com/x/react-charts/styling/#custom-overlay) so you should be able to render any SVG component inside your chart. So the [image component](https://developer.mozilla.org/en-US/docs/Web/SVG/Element/image) should help.

What we could do is to provide a Skeleton component for the bar chart with similar API to the [MUI Skeleton](https://mui.com/material-ui/react-skeleton/)

**Comment by JCQuintas** [comment]
This is out of scope. Custom overlays already accept any React node, so animated formats work without library support.

The general animation story is covered at https://mui.com/x/react-charts/animation/
